pkg/controller/controller.go

```diff
@@ -110,26 +110,45 @@ type controller struct {
 	brokerLister           listers.BrokerLister
 	serviceClassLister     listers.ServiceClassLister
 	instanceLister         listers.InstanceLister
+	bindingLister          listers.BindingLister
+	brokerQueue            workqueue.RateLimitingInterface
+	serviceClassQueue      workqueue.RateLimitingInterface
+	instanceQueue          workqueue.RateLimitingInterface
+	bindingQueue           workqueue.RateLimitingInterface
 }
 
 // Run runs the controller until the given stop channel can be read from.
-func (c *controller) Run(stopCh <-chan struct{}) {
+func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	defer runtimeutil.HandleCrash()
+	defer func() {
+		c.brokerQueue.ShutDown()
+		c.serviceClassQueue.ShutDown()
+		c.instanceQueue.ShutDown()
+		c.bindingQueue.ShutDown()
+	}()
+
 	glog.Info("Starting service-catalog controller")
 
+	for i := 0; i < workers; i++ {
+		go wait.Until(c.brokerWorker, time.Second, stopCh)
+		go wait.Until(c.serviceClassWorker, time.Second, stopCh)
+		go wait.Until(c.instanceWorker, time.Second, stopCh)
+		go wait.Until(c.bindingWorker, time.Second, stopCh)
```

Comment:
the `ShutDown()` calls in the defer block before this do the queue cleanup.  `wait.Until()` is just a way to say, if this goroutine exits for whatever reason (only reason in this case is a program error), restart it.  `wait.Until()` calls the function synchronously so if the function doesn't return, it will not run it again.